### ISSUE TITLE
Game crashes anytime I have Mob Effects in mod folder.

### ISSUE THREAD

**Issue by archaicking** [OreCruncher/MobEffects#2]
---- Minecraft Crash Report ----
// I let you down. Sorry :(

Time: 1/23/20 8:11 PM
Description: Unexpected error

java.lang.NullPointerException: Unexpected error
	at svenhjol.charm.base.CharmAsmHooks.handleMusicTick(CharmAsmHooks.java:70) ~[?:1.14.4-1.4.1] {re:classloading}
	at net.minecraft.client.audio.MusicTicker.func_73660_a(SourceFile) ~[?:?] {re:classloading,xf:fml:charm:MusicTickerTick,xf:fml:charm:MusicTickerStop,xf:fml:charm:MusicTickerPlaying}
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1351) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:869) [?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:384) [?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(SourceFile:155) [?:?] {re:classloading}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_51] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_51] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_51] {}
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_51] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:56) [forge-1.14.4-28.1.92.jar:28.1] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider$$Lambda$391/550707.call(Unknown Source) [forge-1.14.4-28.1.92.jar:28.1] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-4.1.0.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-4.1.0.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-4.1.0.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:81) [modlauncher-4.1.0.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:65) [modlauncher-4.1.0.jar:?] {}

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.14.4
	Minecraft Version ID: 1.14.4
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 3089272320 bytes (2946 MB) / 4160749568 bytes (3968 MB) up to 4294967296 bytes (4096 MB)
	CPUs: 4
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx4G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	ModLauncher: 4.1.0+62+5bfa59b
	ModLauncher launch target: fmlclient
	ModLauncher naming: srg
	ModLauncher services: 
		/eventbus-1.0.0-service.jar eventbus PLUGINSERVICE 
		/forge-1.14.4-28.1.92.jar object_holder_definalize PLUGINSERVICE 
		/forge-1.14.4-28.1.92.jar runtime_enum_extender PLUGINSERVICE 
		/accesstransformers-1.0.1-milestone.0.1+94458e7-shadowed.jar accesstransformer PLUGINSERVICE 
		/forge-1.14.4-28.1.92.jar capability_inject_definalize PLUGINSERVICE 
		/forge-1.14.4-28.1.92.jar runtimedistcleaner PLUGINSERVICE 
		/OptiFine_1.14.4_HD_U_F4.jar OptiFine TRANSFORMATIONSERVICE 
		/forge-1.14.4-28.1.92.jar fml TRANSFORMATIONSERVICE 
	FML: 28.1
	Forge: net.minecraftforge:28.1.92
	FML Language Providers: 
		javafml@28.1
		minecraft@1
	Mod List: 
		forge-1.14.4-28.1.92-client.jar Minecraft {minecraft@1.14.4 NONE}
		pandoras_creatures-1.14.4-0.12.0.jar Pandoras Creatures {pandoras_creatures@1.14.4-0.12.0 NONE}
		ToolBelt-1.14.4-1.12.5.jar Tool Belt {toolbelt@1.12.5 NONE}
		LetSleepingDogsLie-1.14.4-1.1.0.jar Let Sleeping Dogs Lie {dogslie@1.1.0 NONE}
		vilnames_1.14-1.4.jar Villager Names {vilnames@1.4 NONE}
		jei-1.14.4-6.0.0.27.jar Just Enough Items {jei@6.0.0.27 NONE}
		enchantable-1.1.0-mc1.14.4.jar Enchantable {enchantable@1.1.0 NONE}
		Druidcraft-1.14-0.2.32.jar Druidcraft {druidcraft@0.2.32 NONE}
		petyourwolf-1.2.1-1.14.4.jar Pet Your Wolf {petyourwolf@1.2.1 NONE}
		MobEffects-1.14.4-0.0.4.0.jar §3Dynamic Surroundings
§eMob Effects {mobeffects@1.14.4-0.0.4.0 NONE}
		Waystones_1.14.4-5.1.0.jar Waystones {waystones@5.1.0 NONE}
		sit-1.14.4-v1.2.3.jar Sit {sit@v1.2.3 NONE}
		NaturesCompass-1.14.4-1.8.3.jar Nature's Compass {naturescompass@1.14.4-1.8.3 NONE}
		backpacked-1.2.0-mc1.14.4.jar Backpacked {backpacked@1.2.0 NONE}
		obfuscate-0.3.3-1.14.4.jar Obfuscate {obfuscate@0.3.3 NONE}
		DungeonCrawl-1.14.4-1.5.2.jar Dungeon Crawl {dungeoncrawl@1.5.2 NONE}
		u_team_core-1.14.4-2.7.8.138.jar U Team Core {uteamcore@2.7.8.138 NONE}
		furniture-7.0.0-pre15-1.14.4.jar MrCrayfish's Furniture Mod {cfm@7.0.0-pre15 NONE}
		solcarrot-1.14.4-1.9.3.jar Spice of Life: Carrot Edition {solcarrot@1.14.4-1.9.3 NONE}
		Hwyla-forge-1.10.6-B67_1.14.4.jar Waila {waila@version NONE}
		Aquaculture-1.14.4-2.0.2.jar Aquaculture 2 {aquaculture@1.14.4-2.0.2 NONE}
		forgemod_VoxelMap-1.9.13b_for_1.14.4.jar VoxelMap {voxelmap@1.9.13 NONE}
		Charm-1.14.4-1.4.1.jar Charm {charm@1.14.4-1.4.1 NONE}
		forge-1.14.4-28.1.92-universal.jar Forge {forge@28.1.92 NONE}
		Explorercraft-1.14.4-3.2.4.jar Explorercraft {explorercraft@3.2.4 NONE}
		corpse-1.14.4-1.0.3.jar Corpse {corpse@1.14.4-1.0.3 NONE}
		pn_1.14-1.0.jar Pet Names {pn@1.0 NONE}
	Launched Version: 1.14.4-forge-28.1.92
	LWJGL: 3.2.2 build 10
	OpenGL: GeForce GTX 1050/PCIe/SSE2 GL version 4.6.0 NVIDIA 441.08, NVIDIA Corporation
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'forge'
	Type: Client (map_client.txt)
	Resource Packs: vanilla, file/Custom_Walls.zip, file/ChromaHills-128x-1.14-v2.zip
	Current Language: English (US)
	CPU: 4x AMD FX(tm)-4300 Quad-Core Processor 
	OptiFine Version: OptiFine_1.14.4_HD_U_F4
	OptiFine Build: 20191025-153543
	Render Distance Chunks: 11
	Mipmaps: 0
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: BSL_v7.1.04.1.zip
	OpenGlVersion: 4.6.0 NVIDIA 441.08
	OpenGlRenderer: GeForce GTX 1050/PCIe/SSE2
	OpenGlVendor: NVIDIA Corporation
	CpuCount: 4

**Comment by OreCruncher** [comment]
You do not have Sound Control.  It is a required dependency for MobEffects.  Not sure how this plays into the error stack.  The stack indicates that an ASM hook from the Charm mod is not handling a condition thus causing the NPE.

**Comment by archaicking** [comment]
So the Charm mod is really the problem? Cause I did try with Sound Control
and MobEffects which still crashed.

**Comment by OreCruncher** [comment]
Yes it appears to be Charm:
```
java.lang.NullPointerException: Unexpected error
at svenhjol.charm.base.CharmAsmHooks.handleMusicTick(CharmAsmHooks.java:70) ~[?:1.14.4-1.4.1] {re:classloading}
```
It is not checking for a null in it's handleMusicTick() method.

The most current version of Charm on CurseForge is 1.4.3.  Try updating to that.  ~~I looked at the source code in their 1.14 branch did not find the method that is being called so it looks like it was removed.~~ (It was in the master branch.)  If the newest one doesn't work I recommend opening an issue on their github repo.
